.github/workflows/pull-request.yml

```diff
@@ -25,12 +25,12 @@ jobs:
     - uses: actions/setup-python@v2
       with:
         python-version: '3.x'
-#    - name: Cache Linux toolchain
-#      id: cache-linux
-#      uses: actions/cache@v2
-#      with:
-#        path: ./tools/dist
-#        key: key-linux-toolchain
+    - name: Cache Linux toolchain
+      id: cache-linux
+      uses: actions/cache@v2
+      with:
+        path: ./tools/dist
+        key: key-linux-toolchain
```

Comment:
iirc, changing 'key' will force it to use a different cache
it will be expired at some point, but it is not as clear as something like travis (rip)

Reply:
Thx.  I think generally we're OK and the un-done patch here was just a stopgap during development.

The original commenting-out was because the releases have a datestamp in them which ends up making all builds in a day conflict.  So get.py would find gcc-20121101.tgz(from 9AM) in the cached dir and not try and d/l the newer one gcc-20121101.tgz(from 11AM).  The cached version's MD5 would mismatch, so the build failed w/o a toolchain.  Easy fix, drop the caching.  W/everything coming from GH.com, anyway, the cache probably doesn't make much difference...

Reply:
True, speeds on github are really good, so cache most of the time takes longer than just fetching the sources directly...

I think it also could be this 'template', making key depend on the .json so it is always new whenever toolchain is modified
```
key: ${{ runner.os }}-${{ hashFiles('package/package_esp8266com_index.template.json') }}
```
via https://github.com/actions/cache#creating-a-cache-key

Reply:
That's an awesome idea!  Right now the cache will grow w/obsolete versions of the tools, but one based on the JSON would fix that problem.

Did you want the PR for this?  OTW I can throw it together quickly.